lib/desktop/session.rb

```diff
@@ -410,6 +407,7 @@ def local?
     end
 
     def save
+      self.dir
```

Comment:
This would read much better if  `find_session_dir` was renamed to, say, `ensure_session_dir` and this call to `dir`was replaced with a call to `ensure_session_dir`.  Also, there isn't any need for the `self` receiver here.